File: app/src/main/java/kztproject/jp/splacounter/AuthFragment.java
package kztproject.jp.splacounter;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.Toast;

import javax.inject.Inject;

import butterknife.Bind;
import butterknife.OnClick;
import kztproject.jp.splacounter.activity.MainActivity;
import kztproject.jp.splacounter.viewmodel.AuthViewModel;

/**
 * Created by k-seito on 2017/07/17.
 */

public class AuthFragment extends Fragment implements AuthViewModel.Callback {

    @Bind(R.id.token_text)
    EditText tokenText;

    private ProgressDialog progressDialog;

    @Inject
    AuthViewModel viewModel;

    public static AuthFragment newInstance() {
        Bundle args = new Bundle();
        AuthFragment fragment = new AuthFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.activity_auth, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        ((MyApplication) getActivity().getApplication()).component().inject(this);
        viewModel.setCallback(this);
    }

    @OnClick(R.id.login_button)
    public void clickLogin(View view) {
        if (tokenText.length() == 0) {
            Toast.makeText(getContext(), "APIトークンを入力してください", Toast.LENGTH_SHORT).show();
        } else {
            viewModel.login(tokenText.getText().toString());
        }

    }

    @Override
    public void showProgressDialog() {
        if (progressDialog == null) {
            progressDialog = new ProgressDialog(getActivity());
            progressDialog.setMessage("Now Loading...");
        }
        progressDialog.show();
    }

    @Override
    public void dismissProgressDialog() {
        progressDialog.dismiss();
    }

    @Override
    public void loginSuccessed() {
        Toast.makeText(getActivity(), "ログインに成功しました", Toast.LENGTH_SHORT).show();
        startActivity(new Intent(getActivity(), MainActivity.class));

        //TODO call replace fragment
    }

    @Override
    public void loginFailed(Throwable e) {
        Toast.makeText(getActivity(), "ログインに失敗しました", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/kztproject/jp/splacounter/auth/ui/AuthFragment.kt
package kztproject.jp.splacounter.auth.ui

import android.app.ProgressDialog
import android.os.Bundle
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import dagger.android.support.AndroidSupportInjection
import kztproject.jp.splacounter.R
import kztproject.jp.splacounter.databinding.FragmentAuthBinding
import kztproject.jp.splacounter.reward.list.ui.RewardFragment
import kztproject.jp.splacounter.ui_common.replaceFragment
import javax.inject.Inject

class AuthFragment : Fragment(), AuthViewModel.Callback {

    private var progressDialog: ProgressDialog? = null
    private var binding: FragmentAuthBinding? = null

    @Inject
    lateinit var viewModel: AuthViewModel

    companion object {

        fun newInstance(): AuthFragment {
            val args = Bundle()
            val fragment = AuthFragment()
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidSupportInjection.inject(this)
        super.onCreate(savedInstanceState)
        viewModel.setCallback(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentAuthBinding.inflate(inflater, container, false)
        binding!!.viewModel = viewModel
        return binding!!.root
    }

    override fun showProgressDialog() {
        if (progressDialog == null) {
            progressDialog = ProgressDialog(activity)
            progressDialog!!.setMessage("Now Loading...")
        }
        progressDialog!!.show()
    }

    override fun dismissProgressDialog() {
        progressDialog!!.dismiss()
    }

    override fun signUpSucceeded() {
        Toast.makeText(context, "Signed up!", Toast.LENGTH_SHORT).show()

        activity?.replaceFragment(R.id.container, RewardFragment.newInstance())
    }

    override fun loginSucceeded() {
        Toast.makeText(activity, R.string.login_succeeded, Toast.LENGTH_SHORT).show()

        activity?.replaceFragment(R.id.container, RewardFragment.newInstance())
    }

    override fun loginFailed(e: Throwable) {
        Toast.makeText(activity, R.string.login_failed, Toast.LENGTH_SHORT).show()
    }

    override fun showError(@StringRes stringId: Int) {
        Toast.makeText(context, getString(stringId), Toast.LENGTH_SHORT).show()
    }
}